tests/providers/test_gandi.py

```diff
@@ -0,0 +1,24 @@
+# Test for one implementation of the interface
+from lexicon.providers.gandi import Provider
+from lexicon.common.options_handler import env_auth_options
+from integration_tests import IntegrationTests
+from unittest import TestCase
+import pytest
+
+# Hook into testing framework by inheriting unittest.TestCase and reuse
+# the tests which *each and every* implementation of the interface must
+# pass, by inheritance from define_tests.TheTests
+class GandiProviderTests(TestCase, IntegrationTests):
+
+    Provider = Provider
+    provider_name = 'gandi'
+    domain = 'reachlike.ca'
+
+    def _test_options(self):
+        cmd_options = env_auth_options(self.provider_name)
```

Comment:
env_auth_options don't really make sense on a build server. You should be able to load the default test_options and then add any overrides you need. https://github.com/AnalogJ/lexicon/blob/82fa5056df2122357af7f9bec94aebc58b247f91/tests/providers/test_powerdns.py#L18